File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/model/Comment.java
package jp.co.pluswing.mysqlsample.sample.model;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "comments")
public class Comment {

    public Comment() {
    }

    public Comment(Post post) {
        createdAt = new Date();
        this.post = post;
    }

    @Id
    @GeneratedValue
    public Long id;

    @Column(nullable = false)
    public String name;

    @Lob
    @Column(nullable = false, length = 65535)
    public String content;

    @Column(name = "created_at", nullable = false)
    public Date createdAt;

    @ManyToOne
    @JoinColumn(name = "post_id", nullable = false)
    public Post post;
}

File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/model/Comment.kt
package jp.co.pluswing.mysqlsample.sample.model

import java.util.*
import javax.persistence.*

@Entity
@Table(name = "comments")
class Comment {
    constructor() {}
    constructor(post: Post) {
        createdAt = Date()
        this.post = post
    }

    @Id
    @GeneratedValue
    var id: Long? = null

    @Column(nullable = false)
    var name: String = ""

    @Lob
    @Column(nullable = false, length = 65535)
    var content: String = ""

    @Column(name = "created_at", nullable = false)
    var createdAt: Date = Date()

    @ManyToOne
    @JoinColumn(name = "post_id", nullable = false)
    var post: Post = Post()
}